### PR TITLE
Switch to a cloudfront function for the www redirect

### PR DESCRIPTION
**Pull request by cat5inthecradle**
This is a faster and more sustainable option. Using lambda@edge was leading to challenges around lambda versioning and consistently making changes. This implementation will update the live function code whenever the code in the stack is edited.

One loss is that I'm not translating query strings anymore. It's possible if we think it's necessary, but will take some time because we've got limited javascript features here.

## Links

Jira Ticket: INF-506

## Testing story

Thoroughly tested with a standalone stack. We can also easily test the function's code from the AWS Console.

POC Code: https://gist.github.com/cat5inthecradle/9a1186bf07ed07de07b063f1b0eae04c

POC Test: https://console.aws.amazon.com/cloudfront/v3/home?region=us-east-1#/functions/www_redirect?tab=build

## Follow-up work

Delete the lambda function the day after this goes live, once the lambda@edge replicas have have time to be removed.

## Caching

This does not cache function responses, every request to www.code.org will result in a function execution. The redirected domain is likely cached in the main cloudfront distribution.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by cat5inthecradle** [comment]
It's for real going to be done this time.

File: aws/cloudformation/components/www_redirect.yml.erb [hunk 1]
```diff
@@ -81,6 +75,33 @@
...
+          return {
+            statusCode: 302,
+            statusDescription: '302 Redirect to root domain',
+            headers:{
+              "location": { "value": newUrl }
+            }
+          };
+        };
+
   <%=app%>WwwRedirectLambda:
```
Comment (cat5inthecradle) [inline aws/cloudformation/components/www_redirect.yml.erb:102]: We can't delete this function until it's replicas have been removed. This PR will remove the reference, allowing the replicas to delete themselves over the next few hours.

File: aws/cloudformation/components/www_redirect.yml.erb [hunk 2]
```diff
@@ -81,6 +75,33 @@
...
+      AutoPublish: true
+      FunctionCode: !Sub |
+        function handler(event) {
+          var newUrl = 'https://<%=CDO.pegasus_hostname%>';
+
+          // Append path
+          newUrl += event.request.uri;
+
+          // Append query string
+          // - skipping because it's annoying to parse.
```
Comment (Hamms) [inline aws/cloudformation/components/www_redirect.yml.erb]: stray comments?

```suggestion
```

**Review by Hamms — APPROVED** [review]
I'm not particularly familiar with this space, but what I do understand looks good to me!

File: aws/cloudformation/components/www_redirect.yml.erb (hunk 2 above)
Comment (cat5inthecradle) [reply aws/cloudformation/components/www_redirect.yml.erb]: I wasn't sure if it was worth acknowledging that we were not including query params.

**Pull request merged**